transport/transport.go

```diff
@@ -330,12 +330,20 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 	return newHTTP2Server(conn, maxStreams, authInfo)
 }
 
+// DefaultDialerFn is the default dialer.
+func DefaultDialerFn(addr string, timeout time.Duration, cancel <-chan struct{}) (net.Conn, error) {
```

Comment:
This doesn't make an iota of sense; the important part of these functions is that they abstract over the Go version in use. If you ask that I re-write these functions in the tests, that's fine, but it means that these tests will also require build tags to achieve compatibility with Go 1.5 and Go 1.6. Do you also expect users to write Go-version-specific code to make use of this feature?